_studio/mfx_lib/encode/mjpeg/src/mfx_mjpeg_encode.cpp

```diff
@@ -1102,13 +1104,12 @@ mfxStatus MFXVideoENCODEMJPEG::Init(mfxVideoParam *par_in)
     mfxStatus st = MFX_ERR_NONE, QueryStatus = MFX_ERR_NONE;
     mfxVideoParam* par = par_in;
 
-    if(m_isInitialized)
-        return MFX_ERR_UNDEFINED_BEHAVIOR;
+    MFX_CHECK(m_isInitialized, MFX_ERR_UNDEFINED_BEHAVIOR);
+    MFX_CHECK(par == NULL, MFX_ERR_NULL_PTR);
 
-    if(par == NULL)
-        return MFX_ERR_NULL_PTR;
+    MFX_CHECK(mfxInit() == ippStsNoErr, MFX_ERR_UNSUPPORTED);
 
-    MFX_CHECK(CheckExtBuffers(par->ExtParam, par->NumExtParam)== MFX_ERR_NONE, MFX_ERR_INVALID_VIDEO_PARAM);
+    MFX_CHECK(CheckExtBuffers(par->ExtParam, par->NumExtParam) == MFX_ERR_NONE, MFX_ERR_INVALID_VIDEO_PARAM);
```

Comment:
same as above. And that's a slipover from old times since we had this problem before.

Reply:
Done.